util/get_holes.py

```diff
@@ -0,0 +1,58 @@
+import json
+from pathlib import Path
+import requests
+from time import sleep
+
+base = 'https://xivapi.com/'
+spot = 'FishingSpot'
+columns = 'columns=ID,PlaceName.NameNoArticle,Item0.Singular,Item1.Singular,Item2.Singular,Item3.Singular,Item4.Singular,Item5.Singular,Item6.Singular,Item7.Singular,Item8.Singular,Item9.Singular'
+page = 1
+
+response = requests.get(f'{base}{spot}?{columns}&page={page}').json()
+
+results = response['Results']
+
+while (response['Pagination']['Page'] != response['Pagination']['PageTotal']):
+    page += 1
+    response = requests.get(f'{base}{spot}?{columns}&page={page}').json()
+
+    results += response['Results']
+
+# have full list of results now
+holes = {}
+
+for result in results:
```

Comment:
By no means a Python guru, so feel free to shoot me down -- but I think this results in the same code with less `continue`s; does Python provide an advantage to using continue as opposed to failing an if check when iterating through loops?
```suggestion
for result in results:
    # Skip spots without place names
    if result['PlaceName']['NameNoArticle']:
        items = []

        for item in range(10):
            if result['Item'+str(item)]['Singular']:
                items.append(result['Item'+str(item)]['Singular'])

        if items:
            holes[result['PlaceName']['NameNoArticle']] = items
```

Reply:
The reason I use continue here is preference, and for more cognitive reasons than for speed optimizations. It could be faster, I've never actually tested it.

In general, I try to avoid situations where large blocks of code are running inside an if statement, especially when the thing running in the if statement is the more "normal" execution path. When reading code, it requires you to hold in your head the reason for the indentation, and if you encounter an "else" later, you have to remember where it came from. By having an if statement with a continue up top, it concludes that thought immediately, and you can read the rest of the code without holding it in your mind. It doesn't matter a lot in this case, but I've seen a lot of code like 
```
def get_d():
    a = do_thing_1()
    if not a.error:

        b = do_thing_2(a)
        if not b.error:

            c = do_thing_3(b)
            if not c.error:

                d = do_thing_4(c)
                if not d.error:

                    return d
    return false
```

and I find that much harder to read than this:
```
def get_d():
    a = d_thing_1()
    if a.error:
        return false

    b = do_thing_2(a)
    if b.error:
        return false

    c = do_thing_3(b)
    if c.error:
        return false

    d = do_thing_4(c)
    if d.error:
        return false

    return d
```

I'm in the process of rewriting a lot of this code though, in order to get all of the localized names, so it could change in the next commit. That's my motivation for using continue rather than encasing the whole thing in an if block, though.

Reply:
Got it; I saw the code block and had the opposite reaction: "Okay, if not, then... wait, nevermind." So I saw it as a function call that could be broken down into less lines; but, who's to say that someone comes across my code block and doesn't have the immediate reaction you get?

I can see the "return as early as possible" paradigm extending to `continue`. My question initial stemmed from seeing `continue` used frequently in the wild, specifically with Python, so I was curious if there was a difference or minor optimization I hadn't been made aware of. Readability and lessened cognitive complexity are definitely implicit optimizations though for any codebase that needs to be maintained.

Thank you for taking the time to break down your thinking!